src/api/helpers.ts

```diff
@@ -135,8 +135,24 @@ export async function handleGeneratedApiResponse(response: Response): Promise<an
   }
 
   if (!response.ok) {
-    throw new Error(_(t`Échec de la requête ${response.url}, code: ${response.status}`))
+    throw new ApiError(
+      response.status,
+      await response.json(),
+      _(t`Échec de la requête ${response.url}, code: ${response.status}`)
+    )
   }
 
   return await response.json()
 }
+
+export class ApiError extends Error {
+  name = 'ApiError'
+  content: string
```

Comment:
Vu que `response.json()` renvoie `Promise<any>`:
```suggestion
  content: any
```

Reply:
Mon IDE n'aime pas trop le any, du coup je préfère conserver `string` sauf si vous avez une autre idée

Reply:
L'IDE en tant qu'argument psd trop fan non plus héhé ! Si content ne prend pas d'autre valeur que string alors garder string, sinon, demander au backend les autres valeurs possibles.

Reply:
@DanielBaudry il faut que tu ignores le warning avec ts-ignore.
Dans ce cas, le seul type possible est `any`.
Voir https://github.com/pass-culture/pass-culture-app-native/blob/b626e3112de6343161dd625e9876bed8a1b3053b/src/api/helpers.ts#L128-L131 pour le même cas.

Reply:
On ne peut pas typer toutes les réponses du backend, d'où le `any`.

Reply:
Ok je fais ça alors